File: QProcess.Win64/QProcess_Win64.cpp
//+source dllmain.cpp
//+source QProcess_Win64_wrap.cpp
#include "QProcess_Win64.hpp"
#include <QtCore>
#include "strconvQt2.h"
#include "process_util.h"
#include "wstrutil.h"

int Execute3(const std::wstring &exe, const std::wstring &args, const std::wstring &cwd)
{
    std::vector<std::wstring> split = strutil::split(args, std::wstring(L"\t"));
    QStringList arguments;
    for (std::size_t i=0; i<split.size(); i++)
    {
        //qDebug() << i << wide_to_qstr(split[i]);
        arguments.append(wide_to_qstr(split[i]));
    }
    return ProcessUtil::run(false, wide_to_qstr(exe), arguments, wide_to_qstr(cwd));
}

int Execute4(bool windowed, const std::wstring &exe, const std::wstring &args, const std::wstring &cwd)
{
    std::vector<std::wstring> split = strutil::split(args, std::wstring(L"\t"));
    QStringList arguments;
    for (std::size_t i=0; i<split.size(); i++)
    {
        //qDebug() << i << wide_to_qstr(split[i]);
        arguments.append(wide_to_qstr(split[i]));
    }
    return ProcessUtil::run(windowed, wide_to_qstr(exe), arguments, wide_to_qstr(cwd));
}

void Launch3(const std::wstring &exe, const std::wstring &args, const std::wstring &cwd)
{
    std::vector<std::wstring> split = strutil::split(args, std::wstring(L"\t"));
    QStringList arguments;
    for (std::size_t i=0; i<split.size(); i++)
    {
        //qDebug() << i << wide_to_qstr(split[i]);
        arguments.append(wide_to_qstr(split[i]));
    }
    ProcessUtil::launch(true, wide_to_qstr(exe), arguments, wide_to_qstr(cwd));
}

File: QProcess.Win64/QProcess.Win64.cs
//+#inc QProcess.Win64.swig.cs
//+#nuget EasyObject
using System;

namespace QProcess;

public class Win64
{
    public static int Execute(string exe, string[] args, string? cwd = null)
    {
        if (cwd == null) cwd = "";
        Global.EasyObject.Log(new { exe = exe, args = args, cwd = cwd });
        return QProcess_Win64.Execute3(exe, string.Join("\t", args), cwd);
    }
    public static void Launch(string exe, string[] args, string? cwd = null)
    {
        if (cwd == null) cwd = "";
        Global.EasyObject.Log(new { exe = exe, args = args, cwd = cwd });
        QProcess_Win64.Launch3(exe, string.Join("\t", args), cwd);
        //QProcess_Win64.Execute4(true, exe, string.Join("\t", args), cwd);
    }
    public static int Run(bool windowed, string exe, string[] args, string? cwd = null)
    {
        if (cwd == null) cwd = "";
        Global.EasyObject.Log(new { exe = exe, args = args, cwd = cwd });
        int exitCode = QProcess_Win64.Execute4(windowed, exe, string.Join("\t", args), cwd);
        return exitCode;
    }
    public static int RunAndPause(bool windowed, string exe, string[] args, string? cwd = null)
    {
        if (cwd == null) cwd = "";
        Global.EasyObject.Log(new { exe = exe, args = args, cwd = cwd });
        int exitCode = QProcess_Win64.Execute4(windowed, exe, string.Join("\t", args), cwd);
        Console.Error.Write("Press ENTER to continue: ");
        Console.In.ReadLine();
        return exitCode;
    }
}
